failsafe/circuit_breaker.py

```diff
@@ -134,6 +141,27 @@ def get_name(self):
         return 'open'
 
 
+class _HalfOpenState:
+    """
+    A status class representing the half open state of a CircuitBreaker
```

Comment:
Would be nice to like that folwler's article with explanation what it is a half open state?

Reply:
Added more explanation in the comments.